src/ansi-c/c_typecheck_type.cpp

```diff
@@ -89,7 +90,9 @@ void c_typecheck_baset::typecheck_type(typet &type)
   else if(type.id()==ID_typeof)
     typecheck_typeof_type(type);
   else if(type.id()==ID_symbol)
-    typecheck_symbol_type(type);
+    typecheck_symbol_type(to_symbol_type(type));
+  else if(type.id() == ID_typedef_type)
```

Comment:
Per [coding standards](https://github.com/diffblue/cbmc/blob/develop/CODING_STANDARD.md#cprover-conventions) use `can_cast_typet<typedef_typet>(type)` to hide direct id access.